src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policies.erl

```diff
@@ -67,7 +67,33 @@ malformed_request_message(Any, _Req, _State) ->
     error({unexpected_malformed_request_message, Any}).
 
 to_json(Req, #base_state{chef_db_context = DbContext,
-                         resource_state = StubRec
+                         organization_guid = OrgId,
+                         reqid = ReqId
                         } = State) ->
-    Names = chef_db:list(StubRec, DbContext),
-    {chef_json:encode(lists:sort([ hd(NameAsList) ||NameAsList <- Names])), Req, State}.
+    case chef_db:list_all_policy_revisions_by_orgid(DbContext, OrgId) of
+        {error, Why} ->
+            Report = {list_all_policy_revisions_by_orgid, {Why, ReqId}},
+            lager:error("~p", [Report]),
+            error(Report);
+        AllRevisions ->
+            BaseEJSON = build_base_policy_list_ejson(Req, AllRevisions),
+            PolicyEJSON = build_policy_list_ejson(AllRevisions, BaseEJSON),
+            {chef_json:encode(PolicyEJSON), Req, State}
+    end.
+
+build_base_policy_list_ejson(Req, PolicyRevisions) ->
+    RouteFun = oc_chef_wm_routes:bulk_route_fun(policy, Req),
+    UriMap= [{Name, RouteFun(Name)} || {Name, _Rev} <- PolicyRevisions],
+    build_nested_base_ejson(UriMap, {[]}).
+
+build_nested_base_ejson([{PolicyName, URI}|Rest], EJSON) ->
+    NewEJSON = ej:set_p({PolicyName, "uri"}, EJSON, URI),
+    build_nested_base_ejson(Rest, NewEJSON);
+build_nested_base_ejson([], EJSON) ->
+    EJSON.
+
+build_policy_list_ejson([{PolicyName, RevisionID}|Rest], EJSON) ->
+    NewEJSON = ej:set_p({PolicyName, "revisions", RevisionID}, EJSON, {[]}),
+    build_policy_list_ejson(Rest, NewEJSON);
```

Comment:
I think I commented on this in an earlier iteration, but won't this only ever capture one list entry? Should we get the current list if it exists and append to it, then use that as the value?